components/web_discovery/renderer/blink_document_extractor.h

```diff
@@ -0,0 +1,47 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_WEB_DISCOVERY_RENDERER_BLINK_DOCUMENT_EXTRACTOR_H_
+#define BRAVE_COMPONENTS_WEB_DISCOVERY_RENDERER_BLINK_DOCUMENT_EXTRACTOR_H_
+
+#include <vector>
+
+#include "brave/components/web_discovery/common/web_discovery.mojom.h"
+#include "content/public/renderer/render_frame.h"
+#include "content/public/renderer/render_frame_observer.h"
+#include "mojo/public/cpp/bindings/receiver.h"
+#include "services/service_manager/public/cpp/binder_registry.h"
+
+namespace web_discovery {
+
+// Extracts attributes from the current page
```

Comment:
strictly speaking "the current page", "blink document" and "frame" (which this class observes and attaches to) are all different things. So maybe rename this to "WebDiscoveryExtractor" (as we already have PageContentExtractor for ai)?

Reply:
we may have a separate implementation of `mojom::DocumentExtractor` for iOS which would not use blink. it may be worth keeping the name to provide distinction between the two implementations. wdyt?